### PR TITLE
wprs: Fix missing runtime py dep.

### PR DESCRIPTION
**Pull request by ccdunder**
wprs command requires psutils but is not built to have it.

Without this change, running wprs output:
```
Traceback (most recent call last):
  File "/home/ccd/.nix-profile/bin/wprs", line 16, in <module>
    import psutil
ModuleNotFoundError: No module named 'psutil'
```

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mksafavi** [comment]
Hi.
Did you get the `ModuleNotFoundError: No module named 'psutil'` error when you installed it?

For testing the packages I usually run this: 
```nix
nix-shell -E 'with import <nixpkgs> {}; callPackage ./pkgs/by-name/wp/wprs/package.nix {}'
```
I didn't get the error in shell but running the bin file directly gives the error.

your patch fixes that 👍

**Review by mksafavi — APPROVED** [review]

File: pkgs/by-name/wp/wprs/package.nix [hunk 1]
```diff
@@ -41,6 +50,13 @@ rustPlatform.buildRustPackage rec {
 
   preFixup = ''
     cp  wprs "$out/bin/wprs"
+
+    # wprs is a raw python script that needs a module.
+    wrapPythonProgramsIn "$out/bin" "${pythonPath}"
+  '';
+
+  passthru.tests.sanity = runCommand "wprs-sanity" { nativeBuildInputs = [ wprs ]; } ''
+    ${wprs}/bin/wprs -h > /dev/null && touch $out
```
Comment (Aleksanaa) [inline pkgs/by-name/wp/wprs/package.nix:59]: Would be possible to run `testers.testVersion`?

**Comment by Aleksanaa** [comment]
> For testing the packages I usually run this:
> 
> ```nix
> nix-shell -E 'with import <nixpkgs> {}; callPackage ./pkgs/by-name/wp/wprs/package.nix {}'
> ```

Hey but

image

This is making me difficult.

File: pkgs/by-name/wp/wprs/package.nix [hunk 2]
```diff
@@ -41,6 +50,13 @@ rustPlatform.buildRustPackage rec {
 
   preFixup = ''
     cp  wprs "$out/bin/wprs"
+
+    # wprs is a raw python script that needs a module.
+    wrapPythonProgramsIn "$out/bin" "${pythonPath}"
```
Comment (Aleksanaa) [inline pkgs/by-name/wp/wprs/package.nix:55]: Did you try without this? In theory the shebang will be rewritten to python with the package

**Comment by mksafavi** [comment]
> > For testing the packages I usually run this:
> > ```nix
> > nix-shell -E 'with import <nixpkgs> {}; callPackage ./pkgs/by-name/wp/wprs/package.nix {}'
> > ```
> 
> Hey but
> 
> image
> 
> This is making me difficult.

I could be doing something wrong, but I feel like in some cases there are inconsistencies in how they behaves.
I have noticed that in some packages, the binaries in ./result don't have all the runtime dependencies in place. 
but adding the app into my configuration.nix or running the ./result/bin files from the nix-shell works. So I assumed that was enough.

If it's because of missing wrappers, I can look into it to fix it for the other packages that I added as well. I'm sure we previously merged packages that where giving similar errors when you directly run the bin files.

File: pkgs/by-name/wp/wprs/package.nix (hunk 2 above)
Comment (mksafavi) [reply pkgs/by-name/wp/wprs/package.nix:55]: It works without `wrapPythonProgramsIn "$out/bin" "${pythonPath}"`.
it seems like `python3.pkgs.wrapPython` is not needed either?
I built it again with just adding the python3.withPackages line to buildInputs and it doesn't give the error.

**Comment by mksafavi** [comment]
Could you try this patch?
[0001-wprs-add-withpackages.patch.txt](https://github.com/user-attachments/files/17557624/0001-wprs-add-withpackages.patch.txt)

File: pkgs/by-name/wp/wprs/package.nix (hunk 2 above)
Comment (Aleksanaa) [reply pkgs/by-name/wp/wprs/package.nix:55]: Yes, I would assume that.

File: pkgs/by-name/wp/wprs/package.nix (hunk 1 above)
Comment (ccdunder) [reply pkgs/by-name/wp/wprs/package.nix:59]: Yeah I would prefer `testers.testVersion` but the binary doesn't have a version arg.

File: pkgs/by-name/wp/wprs/package.nix (hunk 2 above)
Comment (ccdunder) [reply pkgs/by-name/wp/wprs/package.nix]: > Did you try without this? In theory the shebang will be rewritten to python with the package

TIL. That works. Thanks!

File: pkgs/by-name/wp/wprs/package.nix [hunk 3]
```diff
@@ -5,9 +5,19 @@
   pkg-config,
   libxkbcommon,
   python3,
-  python3Packages,
+
+  # passthru.tests.sanity
```
[the remark this answers is not in the document]
Comment (ccdunder) [reply pkgs/by-name/wp/wprs/package.nix]: I honestly don't know. I've seen it in other pkgs. Removed it for now.

**Comment by ccdunder** [comment]
@mksafavi @Aleksanaa thanks for the helpful feedback! I think I've incorporated it all.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**